src/components/screens/monitor/delegates/withForgingStatus.js

```diff
@@ -72,35 +72,40 @@ const withForgingStatus = delegatesKey => (ChildComponent) => {
       );
       const height = blocks[0] && blocks[0].height;
       this.setState({
-        nextForgers: nextForgers.reduce((accumulator, delegate, i) => ({
-          ...accumulator,
-          [delegate.publicKey]: {
-            forgingTime: moment().add(i * 10, 'seconds'),
-            nextHeight: height + i + 1,
-          },
-        }), {}),
+        nextForgers: nextForgers.map((delegate, i) => ({
+          ...delegate,
+          forgingTime: moment().add(i * 10, 'seconds'),
+          nextHeight: height + i + 1,
+        })),
       });
     }
 
     componentDidUpdate(prevProps) {
       const { latestBlocks } = this.props;
       const newBlock = latestBlocks[0] || {};
+      const lastForgerIndex = this.state.nextForgers.findIndex(delegate =>
+        delegate.publicKey === newBlock.generatorPublicKey);
+
       if (prevProps.latestBlocks[0] && prevProps.latestBlocks[0].height < newBlock.height) {
-        this.setState({
-          nextForgers: {
-            ...this.state.nextForgers,
-            [newBlock.generatorPublicKey]: {
-              forgingTime: moment().add(voting.numberOfActiveDelegates * 10, 'seconds'),
-              nextHeight: newBlock.leight + voting.numberOfActiveDelegates,
-            },
-          },
-        });
+        const nextForgers = this.getUpdatedNextForgersList(lastForgerIndex);
+        this.setState({ nextForgers });
         if (newBlock.height % 101 === 1) { // to update next forgers in a new round
           this.loadNextForgers(latestBlocks);
         }
       }
     }
 
+    getUpdatedNextForgersList(forgerIndex) {
+      const nextForgers = [...this.state.nextForgers];
+      nextForgers[forgerIndex] = {
+        ...nextForgers[forgerIndex],
+        forgingTime: moment().add(voting.numberOfActiveDelegates * 10, 'seconds'),
```

Comment:
- 🐛 `forgingTime` needs to be updated not only for `nextForgers[forgerIndex]`, but also for all `nextForgers` before `forgerIndex` (if there are any). 

💡 So the update can be performed with `map` on `forgersToPush`, but then the `forgingTime` needs to depend on the index.